### PR TITLE
Implement QueryBalanceDetails RPC method

### PR DESCRIPTION
**Pull request by ZhmakAS**
* Resolves: # 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits are signed
- [ ] Tests included/updated or not needed
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: rpc/client.go [hunk 1]
```diff
@@ -119,6 +119,8 @@ type ClientInformational interface {
 	GetPeers(ctx context.Context) (InfoGetPeerResult, error)
 	// QueryBalance queries for balances under a given PurseIdentifier
 	QueryBalance(ctx context.Context, identifier PurseIdentifier) (QueryBalanceResult, error)
+	// QueryBalanceDetails query for full balance information using a purse identifier and a state identifier
+	QueryBalanceDetails(ctx context.Context, purseIdentifier PurseIdentifier, stateIdentifier BalanceStateIdentifier) (QueryBalanceDetailsResult, error)
```
Comment (davidatwhiletrue) [inline rpc/client.go:123]: @ZhmakAS I see in other RPC methods that the pattern is to have multiple methods to query by block hash, height or latest. I would do the same here.
For `QueryBalance` it's also missing and has been reported in https://github.com/make-software/casper-go-sdk/issues/85.

**Review by davidatwhiletrue — COMMENTED** [review]
@ZhmakAS see my comment

**Review by koltsov-iv — COMMENTED** [review]
Please check my comment.

File: rpc/rpc_client.go [hunk 2]
```diff
@@ -264,6 +264,39 @@ func (c *client) QueryBalance(ctx context.Context, identifier PurseIdentifier) (
...
+func (c *client) QueryBalanceDetailsByBlockHash(ctx context.Context, purseIdentifier PurseIdentifier, blockHash string) (QueryBalanceDetailsResult, error) {
+	var result QueryBalanceDetailsResult
+	return result, c.processRequest(ctx, MethodQueryBalanceDetails, QueryBalanceDetailsRequest{purseIdentifier, BalanceStateIdentifier{
+		Block: &BlockIdentifier{
+			Hash: &blockHash,
+		},
+	}}, &result)
+}
+
+func (c *client) QueryBalanceDetailsByStateRoot(ctx context.Context, purseIdentifier PurseIdentifier, stateRootHash, timestamp string) (QueryBalanceDetailsResult, error) {
```
[the remark this answers is not in the document]
Comment (davidatwhiletrue) [reply rpc/rpc_client.go:290]: There's a more generic `QueryBalanceDetails` method to get balance from latest state if no explicit state is passed. I would make this one work without a `BalanceStateIdentifier` rateher than having a explicit method `QueryBalanceDetailsByStateRoot` having an optional `stateRootHash`. Also, it makes two calls to the node when 1 is enough.

**Review by koltsov-iv — APPROVED** [review]

File: rpc/client.go [hunk 3]
```diff
@@ -45,10 +45,10 @@ type ClientPOS interface {
 // The response should be identical, regardless of the node queried,
 // as the information in question is objective and common to all nodes within a network.
 type ClientInformational interface {
-	// GetAccountBalance returns a purse's balance from a network.
+	// GetBalance returns a purse's balance from a network.
 	// The request takes in the formatted representation of a purse URef as a parameter.
 	// If the param stateRootHash is nil, the client will make an additional RPC call to retrieve the latest stateRootHash.
-	GetAccountBalance(ctx context.Context, stateRootHash *string, purseURef string) (StateGetBalanceResult, error)
+	GetBalance(ctx context.Context, purseURef string, stateRootHash *string) (StateGetBalanceResult, error)
```
Comment (davidatwhiletrue) [inline rpc/client.go:51]: Just realized. Should we have `GetLatestBalance` and `GetBalanceByStateRootHash` methods?

**Review by davidatwhiletrue — APPROVED** [review]
LGTM. Check my comment about `GetBalance` method.

**Pull request merged**